### DISCUSSION TITLE
Modbus connection overwhelms sDongle --> breaks control of master inverter leasing to voltage spikes ?!

### DISCUSSION THREAD

**Discussion by Kugelfang666** [wlcrs/huawei_solar#722] [Setting up the integration] [answered]
Hi all, 

There are long discussions over in the German https://www.photovoltaikforum.com/ regarding issues in a scenario with cascaded inverters and a Luna battery connected to the system. 

Apparently the implementation of Huawei is so that in these scenarios the feedback loops of the inverters are controlled by the sDongle rather than the internal controllers of the inverters. There are a lot of users which are suffering from the feedback loop of the master showing unusual spikes in the string voltage leading to lower production. 

A common hypothesis is that the frequent polling of the sDongle is interfering with its task to control the inverter leading to this issues. 

I described my own issue here including many pictures:
https://www.photovoltaikforum.com/thread/228815-kurze-spannungsabfälle-in-stringspannung-huawei-10ktl/

a longer discussion can also be found here
https://www.photovoltaikforum.com/thread/200153-leistungseinbrüche-huawei-wechselrichter-mit-luna-und-kaskade/

Just posting this here to create some awareness since this integration is being suspected as possible culprit in these scenarios.

![Screenshot 3](https://github.com/wlcrs/huawei_solar/assets/3289654/2a14ef7b-825f-4de9-a2d6-35fd80ba14ed)

**Comment by LotharWoman** [comment]
Yes, I've read about this problem too, but rest assured, it has nothing to do with the query frequency of the sDongle.

**Reply by Kugelfang666** [reply]
Interesting, what makes you think that?

**Comment by LotharWoman** [comment]
I had 2 customers with a similar problem with a cascaded plant. Completely switching off Modbus-TCP on the system did not change the problem.
 I was able to help a customer by relocating the communication cable to the power meter 1m away from the AC lines. For the other customer, this was not structurally possible. The slave was connected there with its own sDongle.

**Reply by Kugelfang666** [reply]
mhm ok but if that would be interference from AC to the Modbus line to the DTSU then why would this only show up in the voltage signal of the master but not the slave inverter?

**Reply by LotharWoman** [reply, marked as answer]
I don't know yet whether it's really due to interference. It is not easy to identify the problem in FusionSolar with certainty. Owner 1 hasn't had several days of continuous sun yet. So I don't know if this will finally eliminate it.
 However, user 2 with two sDongles definitely no longer has the voltage fluctuations.

**Comment by Kugelfang666** [comment]
Huawei support confirmed the issue, I have removed the cascade from my setup and now have two inverters each with their own dongle. Ever since the issue is gone for me
